internal/auth0/email/expand.go

```diff
@@ -0,0 +1,159 @@
+			return stop
+		})
+		return stop
+	})
+}
+
+func expandEmailProviderSendGrid(config cty.Value, emailProvider *management.EmailProvider) {
+	config.GetAttr("credentials").ForEachElement(func(_ cty.Value, credentials cty.Value) (stop bool) {
+		emailProvider.Credentials = &management.EmailProviderCredentialsSendGrid{
+			APIKey: value.String(credentials.GetAttr("api_key")),
+		}
+		return stop
+	})
+}
+
+func expandEmailProviderSparkPost(config cty.Value, emailProvider *management.EmailProvider) {
+	config.GetAttr("credentials").ForEachElement(func(_ cty.Value, credentials cty.Value) (stop bool) {
+		emailProvider.Credentials = &management.EmailProviderCredentialsSparkPost{
+			APIKey: value.String(credentials.GetAttr("api_key")),
+			Region: value.String(credentials.GetAttr("region")),
+		}
+		return stop
+	})
+}
+
+func expandEmailProviderMailgun(config cty.Value, emailProvider *management.EmailProvider) {
+	config.GetAttr("credentials").ForEachElement(func(_ cty.Value, credentials cty.Value) (stop bool) {
+		emailProvider.Credentials = &management.EmailProviderCredentialsMailgun{
+			APIKey: value.String(credentials.GetAttr("api_key")),
+			Domain: value.String(credentials.GetAttr("domain")),
+			Region: value.String(credentials.GetAttr("region")),
+		}
+		return stop
+	})
+}
+
+func expandEmailProviderSmtp(config cty.Value, emailProvider *management.EmailProvider) {
+	config.GetAttr("credentials").ForEachElement(func(_ cty.Value, credentials cty.Value) (stop bool) {
+		emailProvider.Credentials = &management.EmailProviderCredentialsSMTP{
+			SMTPHost: value.String(credentials.GetAttr("smtp_host")),
+			SMTPPort: value.Int(credentials.GetAttr("smtp_port")),
+			SMTPUser: value.String(credentials.GetAttr("smtp_user")),
+			SMTPPass: value.String(credentials.GetAttr("smtp_pass")),
+		}
+		return stop
+	})
+
+	config.GetAttr("settings").ForEachElement(func(_ cty.Value, settings cty.Value) (stop bool) {
+		settings.GetAttr("headers").ForEachElement(func(_ cty.Value, headers cty.Value) (stop bool) {
+			emailProvider.Settings = management.EmailProviderSettingsSMTP{
+				Headers: &management.EmailProviderSettingsSMTPHeaders{
+					XMCViewContentLink:   value.String(headers.GetAttr("x_mc_view_content_link")),
+					XSESConfigurationSet: value.String(headers.GetAttr("x_ses_configuration_set")),
+				},
+			}
+			return stop
+		})
+		return stop
+	})
+}
+
+func expandEmailTemplate(config cty.Value) *management.EmailTemplate {
+	emailTemplate := &management.EmailTemplate{
+		Template:               value.String(config.GetAttr("template")),
+		Body:                   value.String(config.GetAttr("body")),
+		From:                   value.String(config.GetAttr("from")),
+		ResultURL:              value.String(config.GetAttr("result_url")),
+		Subject:                value.String(config.GetAttr("subject")),
+		Syntax:                 value.String(config.GetAttr("syntax")),
+		URLLifetimeInSecoonds:  value.Int(config.GetAttr("url_lifetime_in_seconds")),
```

Comment:
(unrelated to this PR) this `management.EmailTemplate` is misspelled.

Reply:
Ah, that's really unfortunately 😢 We'll have to fix it within the go SDK. Thanks for spotting!